pkg/localstore/gc.go

```diff
@@ -179,6 +179,9 @@ func (db *DB) collectGarbage() (evicted uint64, done bool, err error) {
 
 		storedItem, err := db.retrievalDataIndex.Get(item)
 		if err != nil {
+			if errors.Is(err, leveldb.ErrNotFound) {
```

Comment:
So I had thought about this. But ideally this shouldnt happen as gcIndex is also deleted once it is removed.

Either way, if retrievalDataIndex is not present, then we should atleast deleted the gcIndex so we dont go through this entry again.

Reply:
yes should not happen but we have thousands of "msg"="collect garbage failed" errors